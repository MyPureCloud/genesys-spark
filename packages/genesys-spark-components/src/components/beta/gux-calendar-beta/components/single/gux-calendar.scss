@use '../../../../../style/mixins';

.gux-calendar-beta {
  box-sizing: border-box;
  display: inline-flex;
  flex-direction: column;
  font-family: var(--gse-ui-calendarMenu-month-defaultText-fontFamily);
  font-size: var(--gse-ui-calendarMenu-month-defaultText-fontSize);
  background: var(--gse-ui-calendarMenu-date-selected-foregroundColor);
  border-radius: var(--gse-ui-calendarMenu-single-body-borderRadius);
  // TODO: Get box-shadow token working instead of hardcoding
  // box-shadow: var(--gse-ui-calendarMenu-boxShadow-color)px var(--gse-ui-calendarMenu-boxShadow-x)px var(--gse-ui-calendarMenu-boxShadow-y)px var(--gse-ui-calendarMenu-boxShadow-blur)px var(----gse-ui-calendarMenu-boxShadow-spread)px;
  box-shadow: rgb(34 37 41 / 24%) 0 2px 4px;

  .gux-header {
    display: flex;
    align-items: center;
    justify-content: space-between;
    height: var(--gse-ui-calendarMenu-month-single-header-height);
    padding: var(--gse-ui-calendarMenu-header-padding);
    font-style: normal;
    font-weight: var(--gse-ui-calendarMenu-month-currentText-fontWeight);
    color: var(--gse-ui-calendarMenu-header-foregroundColor);
    text-align: center;
    background-color: var(--gse-ui-calendarMenu-header-backgroundColor);
    border-radius: var(--gse-ui-calendarMenu-single-header-borderRadius);

    .gux-header-month-and-year {
      width: var(--gse-ui-calendarMenu-month-single-header-textWidth);
      font-family: var(--gse-ui-calendarMenu-month-headerText-fontFamily);
      font-size: var(--gse-ui-calendarMenu-month-headerText-fontSize);
      line-height: var(--gse-ui-calendarMenu-month-headerText-lineHeight);
      color: var(--gse-ui-calendarMenu-month-selected-foregroundColor);
    }

    .gux-left,
    .gux-right {
      color: var(--gse-ui-calendarMenu-month-monthHeader-foregroundColor);
      cursor: pointer;
      background: none;
      border: none;
      outline: none;

      &:focus-visible {
        outline: 2px solid #aac9ff;
        outline-offset: 1px;
      }

      gux-icon {
        width: 16px;
        height: 16px;
      }
    }
  }

  .gux-week-days {
    padding-top: 24px;
    padding-right: 24px;
    padding-left: 24px;
    font-family: var(--gse-ui-calendarMenu-day-headerText-fontFamily);
    font-size: var(--gse-ui-calendarMenu-day-headerText-fontSize);
    font-weight: var(--gse-ui-calendarMenu-day-headerText-fontWeight);
    color: var(--gse-ui-calendarMenu-day-dayHeader-foregroundColor);
    text-align: center;

    .gux-week-day {
      display: inline-block;
      width: var(--gse-ui-calendarMenu-day-date-size);
      height: 16px;
      font-weight: var(--gse-ui-calendarMenu-day-header-fontWeight);
    }
  }

  .gux-content {
    padding: var(--gse-ui-calendarMenu-dateBody-padding);
    color: var(--gse-core-color-cornflower-900);
    background-color: var(--gse-ui-calendarMenu-backgroundColor);
    border-radius: var(--gse-ui-calendarMenu-single-body-borderRadius);
  }

  .gux-content-week {
    .gux-content-date {
      display: inline-block;
      width: var(--gse-ui-calendarMenu-day-date-size);
      height: var(--gse-ui-calendarMenu-day-input-height);
      font-family: var(--gse-ui-calendarMenu-date-currentText-fontFamily);
      font-size: var(--gse-ui-calendarMenu-date-defaultText-fontSize);
      font-style: normal;
      font-weight: var(--gse-ui-calendarMenu-date-defaultText-fontWeight);
      line-height: 32px;
      color: var(--gse-core-color-cornflower-600);
      text-align: center;
      vertical-align: middle;
      border: none;
      outline: none;
      opacity: var(--gse-ui-calendarMenu-disabled-opacity);

      &.gux-current-month {
        // TODO: Is this the correct token to use here?
        color: var(--gse-ui-calendarMenu-month-default-foregroundColor);
        opacity: 1;
      }

      &.gux-hidden {
        display: none;
      }

      &.gux-disabled {
        pointer-events: none;
      }

      &.gux-selected {
        color: var(--gse-ui-calendarMenu-date-selected-foregroundColor);
        background-color: var(
          --gse-ui-calendarMenu-date-selected-backgroundColor
        );
        border: none;
        border-radius: var(--gse-ui-calendarMenu-month-borderRadius);
        outline: none;

        &:hover {
          background-color: var(
            --gse-ui-calendarMenu-date-selected-hoverBackgroundColor
          );
        }
      }

      &:hover {
        cursor: pointer;
        background-color: var(--gse-ui-calendarMenu-date-hover-backgroundColor);
        border-radius: var(--gse-ui-calendarMenu-month-borderRadius);
      }

      &:focus-visible {
        border-radius: var(--gse-ui-calendarMenu-month-focusBorderRadius);
        outline: var(--gse-semantic-focusRing-default-border-width) solid
          var(--gse-semantic-focusRing-default-border-color);
      }

      .gux-sr-only {
        @include mixins.gux-sr-only-clip;
      }

      &.gux-current-date {
        font-family: var(--gse-ui-calendarMenu-month-currentText-fontFamily);
        font-size: var(--gse-ui-calendarMenu-date-currentText-fontSize);
        font-weight: var(--gse-ui-calendarMenu-date-currentText-fontWeight);
      }
    }
  }
}
