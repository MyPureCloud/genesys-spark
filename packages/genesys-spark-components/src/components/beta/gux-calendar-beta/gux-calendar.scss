@use '~genesys-spark/dist/beta/tokens/flare/ui.scss';
@use '~genesys-spark/dist/scss/focus.scss';
@use '~genesys-spark/dist/scss/mixins.scss';

.gux-calendar-beta.gux-disabled {
  pointer-events: none;
  opacity: ui.$gse-ui-calendarMenu-disabled-opacity;
}

.gux-calendar-beta {
  box-sizing: border-box;
  display: inline-flex;
  flex-direction: column;
  font-family: ui.$gse-ui-calendarMenu-month-defaultText-fontFamily;
  font-size: ui.$gse-ui-calendarMenu-month-defaultText-fontSize;
  background: ui.$gse-ui-calendarMenu-backgroundColor;
  border-radius: 8px; // --gse-ui-calendarMenu-single-header-borderRadius and --gse-ui-calendarMenu-single-body-borderRadius need to be combined
  box-shadow: ui.$gse-ui-calendarMenu-boxShadow;

  .gux-header {
    display: flex;
    align-items: center;
    justify-content: space-between;
    block-size: ui.$gse-ui-calendarMenu-month-single-header-height;
    padding: ui.$gse-ui-calendarMenu-header-padding;
    font-style: normal;
    font-weight: ui.$gse-ui-calendarMenu-month-currentText-fontWeight;
    color: ui.$gse-ui-calendarMenu-header-foregroundColor;
    text-align: center;
    background-color: ui.$gse-ui-calendarMenu-header-backgroundColor;
    border-radius: ui.$gse-ui-calendarMenu-single-header-borderRadius;

    .gux-header-month-and-year {
      inline-size: ui.$gse-ui-calendarMenu-month-single-header-textWidth;
      font-family: ui.$gse-ui-calendarMenu-month-headerText-fontFamily;
      font-size: ui.$gse-ui-calendarMenu-month-headerText-fontSize;
      line-height: ui.$gse-ui-calendarMenu-month-headerText-lineHeight;
      color: ui.$gse-ui-calendarMenu-month-selected-foregroundColor;
    }

    .gux-left,
    .gux-right {
      block-size: 100%;
      color: ui.$gse-ui-calendarMenu-header-foregroundColor;
      cursor: pointer;
      outline: none;
      background: none;
      border: none;

      &:focus-visible {
        @include focus.gux-focus-ring;
      }
    }
  }

  .gux-content {
    display: flex;
    flex-direction: column;
    gap: ui.$gse-ui-calendarMenu-dateBody-gap;
    padding: ui.$gse-ui-calendarMenu-dateBody-padding;
    color: ui.$gse-ui-calendarMenu-date-default-foregroundColor;
    background-color: ui.$gse-ui-calendarMenu-backgroundColor;
    border-radius: ui.$gse-ui-calendarMenu-single-body-borderRadius;
  }

  .gux-week-days {
    font-family: ui.$gse-ui-calendarMenu-day-headerText-fontFamily;
    font-size: ui.$gse-ui-calendarMenu-day-headerText-fontSize;
    font-weight: ui.$gse-ui-calendarMenu-day-headerText-fontWeight;
    color: ui.$gse-ui-calendarMenu-month-default-foregroundColor;
    text-align: center;

    .gux-week-day {
      display: inline-block;
      inline-size: ui.$gse-ui-calendarMenu-day-range-height;
      block-size: ui.$gse-ui-calendarMenu-day-range-width;
    }
  }
}
