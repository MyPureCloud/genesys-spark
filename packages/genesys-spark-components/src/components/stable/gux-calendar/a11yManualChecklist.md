# gux-calendar manual accessibility testing status

**Last Updated:** Wed Dec 22 2021 14:42:33 GMT-0500 (Eastern Standard Time)
| Pass | WCAG Success Criterion | Notes |
| --- | --- | --- |
| ❌ | [2.1.1 Keyboard](https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html) | COMUI-827 Forward and back arrows should be focusable. Keyboard navigation within the date input field should be more clear to screenreader user |
| ✅ | [2.1.2 No Keyboard Trap](https://www.w3.org/WAI/WCAG21/Understanding/no-keyboard-trap.html) | - |
| ✅ | [2.4.3 Focus Order](https://www.w3.org/WAI/WCAG21/Understanding/focus-order.html) | - |
| ✅ | [2.4.7 Focus Visible](https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html) | - |
| ❌ | [2.5.3 Label in Name](https://www.w3.org/WAI/WCAG21/Understanding/label-in-name.html#dfn-name) | COMUI-827 The day, month and year is read to the screenreader, but it should be reworded in a more natural way |
| ✅ | [3.2.1 On Focus](https://www.w3.org/WAI/WCAG21/Understanding/on-focus.html) | - |
| ✅ | [3.2.2 On Input](https://www.w3.org/WAI/WCAG21/Understanding/on-input.html) | - |
| ✅ | [3.3.1 Error Identification](https://www.w3.org/WAI/WCAG21/Understanding/error-identification.html) | - |
| ✅ | [3.2.2 Labels or Instructions](https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions.html) | - |
| ✅ | [4.1.2: Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html) | - |
| ✅ | [4.1.3 Status Messages](https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html) | - |
