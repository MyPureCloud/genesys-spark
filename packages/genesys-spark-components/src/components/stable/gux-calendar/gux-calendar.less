@import (reference) '../../../style/color.less';
@import (reference) '../../../style/typography.less';
@import (reference) '../../../style/color-palette.less';

// Style
:host {
  display: inline-block;
}

.gux-calendar {
  margin-top: 2px;
  border: 1px solid @gux-grey-30;
  border-radius: 4px;

  .gux-header {
    position: relative;
    display: flex;
    align-items: center;
    justify-content: space-between;
    height: 16px;
    padding: 16px;
    color: @gux-grey-100;
    background-color: @gux-blue-60;
    border-top-left-radius: 3px;
    border-top-right-radius: 3px;

    button {
      position: absolute;
      padding: 5px;
      color: @gux-grey-100;
      cursor: pointer;
      background: none;
      border: none;
      outline: none;

      &:focus-visible {
        outline: @gux-blue-80 auto 5px;
      }

      &.gux-left {
        left: 18px;
      }

      &.gux-right {
        right: 18px;
      }

      gux-icon {
        width: 16px;
        height: 16px;
        pointer-events: none;
      }
    }

    .gux-month-list {
      display: flex;
      justify-content: space-between;
      width: 100%;

      label {
        width: 100%;
        text-align: center;
        .body-font();
        .bold-font();
      }
    }
  }

  .gux-content {
    display: flex;
    align-items: flex-start;
    padding: 20px 24px;
    color: @gux-black-50;
    background-color: @gux-grey-100;
    border-bottom-right-radius: 3px;
    border-bottom-left-radius: 3px;

    table {
      width: 168px;
      text-align: left;
      table-layout: fixed;
      border-spacing: 0;
      -ms-user-select: none;
      user-select: none;

      &:not(:last-child) {
        padding-right: 24px;
      }

      tr {
        height: 28px;

        &:empty {
          display: none;
        }

        th,
        td {
          width: 28px;
          height: 28px;
          padding: 0;
          margin: 0;
          text-align: center;
          .body-font();
        }

        th {
          .bold-font();
        }

        td {
          cursor: pointer;

          &.gux-hovered {
            background-color: @gux-blue-100;
          }

          &:hover:not(&.gux-start-date):not(&.gux-end-date) {
            color: @gux-grey-100;
            background-color: @gux-blue-60;
            border-radius: 4px;
          }

          &.gux-disabled {
            color: fade(@gux-black-50, 50%);
            pointer-events: none;
          }

          &.gux-not-in-month {
            color: fade(@gux-black-100, 50%);

            &:hover {
              color: @gux-black-50;
              background-color: @gux-border;
            }

            &.gux-hidden {
              visibility: hidden;
            }
          }

          &.gux-selected {
            color: @gux-grey-100;
            background-color: @gux-blue-60;
            border-radius: 4px;

            &.gux-start-date {
              border-radius: 4px 0 0 4px;
            }

            &.gux-end-date {
              border-radius: 0 4px 4px 0;
            }
          }
        }
      }
    }
  }
}

.gux-sr-only {
  .gux-sr-only();
}
