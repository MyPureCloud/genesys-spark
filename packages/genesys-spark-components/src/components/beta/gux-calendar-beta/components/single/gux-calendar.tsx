import { Component, Element, h, JSX, State } from '@stencil/core';
import { IWeekElement, GuxCalendarDayOfWeek } from '../../gux-calendar.types';
import { fromIsoDate } from '@utils/date/iso-dates';
import {
  getWeekdays,
  getFirstOfMonth,
  getDateAsMonthYear,
  firstDateInMonth
} from '../../services/calendar.service';
import { getDesiredLocale, getStartOfWeek } from '../../../../../i18n';
import { buildI18nForComponent, GetI18nValue } from '../../../../../i18n';
import translationResources from '../../i18n/en.json';
import { afterNextRenderTimeout } from '@utils/dom/after-next-render';
import { logError } from '@utils/error/log-error';
import simulateNativeEvent from '@utils/dom/simulate-native-event';

@Component({
  styleUrl: 'gux-calendar.scss',
  tag: 'gux-calendar-beta',
  shadow: true
})
export class GuxCalendar {
  @Element()
  root: HTMLElement;

  @State()
  private focusedValue: Date;

  @State()
  private minValue: Date;

  @State()
  private maxValue: Date;

  private locale: string = 'en';

  private i18n: GetI18nValue;

  // Total number of dates that will display for each month in the calendar
  private MONTH_DATE_COUNT: number = 42;

  // Keeps track of the start day of the week based on user's locale
  // Some locales will have the start day of the week be different than others
  private startDayOfWeek: GuxCalendarDayOfWeek;

  private get slottedInput(): HTMLInputElement {
    return this.root.querySelector('input[type="date"]');
  }

  async componentWillLoad(): Promise<void> {
    if (!this.slottedInput) {
      logError(
        this.root,
        'You must slot a date input element like so: input[type="date"].'
      );
    }

    this.locale = getDesiredLocale(this.root);

    // Set start day of week
    const startDayOfWeek = this.slottedInput.getAttribute('start-day-of-week');
    if (startDayOfWeek?.length) {
      this.startDayOfWeek = parseInt(
        startDayOfWeek,
        10
      ) as GuxCalendarDayOfWeek;
    }
    this.startDayOfWeek = this.startDayOfWeek || getStartOfWeek(this.locale);

    this.i18n = await buildI18nForComponent(this.root, translationResources);

    if (this.slottedInput.value) {
      this.focusedValue = fromIsoDate(this.slottedInput.value);
    }

    // Set min value from the "min" input prop
    if (this.slottedInput.min) {
      this.minValue = new Date(this.slottedInput.min);
      this.minValue.setHours(0, 0, 0, 0);
    }
    // Set max value from the "max" input prop
    if (this.slottedInput.max) {
      this.maxValue = new Date(this.slottedInput.max);
      this.maxValue.setHours(0, 0, 0, 0);
    }
  }

  private onDateClick(date: Date): void {
    if (this.isInvalidDate(date)) {
      return;
    }

    this.focusedValue = new Date(date.getTime());
    this.slottedInput.value = date.toISOString().substring(0, 10);
    simulateNativeEvent(this.root, 'input');
  }

  private setDateAfterArrowKeyPress(
    event: KeyboardEvent,
    newDayValue: number
  ): void {
    event.preventDefault();
    this.focusedValue = new Date(
      this.getFocusedValue().getFullYear(),
      this.getFocusedValue().getMonth(),
      this.getFocusedValue().getDate() + newDayValue,
      0,
      0,
      0
    );
    afterNextRenderTimeout(() => {
      this.focusDate();
    });
  }

  /**
   * Focus the focused date
   */
  private focusDate() {
    const target: HTMLTableCellElement = this.root.shadowRoot.querySelector(
      `.gux-content-date[data-date="${this.focusedValue.getTime()}"]`
    );
    if (target) {
      target.focus();
    }
  }

  private onKeyDown(event: KeyboardEvent): void {
    switch (event.key) {
      case ' ':
        break;
      case 'Enter':
        event.preventDefault();
        this.onDateClick(this.getFocusedValue());
        afterNextRenderTimeout(() => {
          this.focusDate();
        });
        break;
      case 'ArrowDown':
        this.setDateAfterArrowKeyPress(event, 7);
        break;
      case 'ArrowUp':
        this.setDateAfterArrowKeyPress(event, -7);
        break;
      case 'ArrowLeft':
        this.setDateAfterArrowKeyPress(event, -1);
        break;
      case 'ArrowRight':
        this.setDateAfterArrowKeyPress(event, 1);
        break;
      case 'PageUp':
        event.preventDefault();
        this.setNewFocusedMonth(1);
        afterNextRenderTimeout(() => {
          this.focusDate();
        });
        break;
      case 'PageDown':
        event.preventDefault();
        this.setNewFocusedMonth(-1);
        afterNextRenderTimeout(() => {
          this.focusDate();
        });
        break;
    }
  }

  private isInvalidDate(date: Date): boolean {
    return (
      (this.minValue && date.getTime() <= this.minValue.getTime()) ||
      (this.maxValue && date.getTime() > this.maxValue.getTime())
    );
  }

  private getMonthDays(): IWeekElement[] {
    const firstOfMonth = getFirstOfMonth(this.getFocusedValue());
    const weeks = [];
    let currentWeek = { dates: [] };
    let weekDayIndex = 0;
    const currentMonth = firstOfMonth.getMonth();
    const selectedValue = this.getSelectedValue();
    const firstOfMonthDate = new Date(firstOfMonth.getTime());
    const currentDate = firstDateInMonth(
      currentMonth,
      firstOfMonthDate.getFullYear(),
      this.startDayOfWeek
    );

    // Generate all of the dates in the current month
    for (let d = 0; d < this.MONTH_DATE_COUNT + 1; d += 1) {
      const selected =
        selectedValue.getTime() === currentDate.getTime() &&
        this.focusedValue !== undefined;

      if (weekDayIndex > 0 && weekDayIndex % 7 === 0) {
        weeks.push(currentWeek);
        currentWeek = {
          dates: []
        };
      }

      // Check if a date is outside the defined date range boundaries
      const disabled = this.isInvalidDate(currentDate);

      const focused =
        this.getFocusedValue()?.getTime() === currentDate.getTime() &&
        this.getFocusedValue()?.getTime() !== selectedValue.getTime(); // Do not show preview value for date if it's already selected

      const today = new Date();

      currentWeek.dates.push({
        date: new Date(currentDate),
        disabled,
        inCurrentMonth: currentMonth === currentDate.getMonth() && !disabled,
        selected: selected && selectedValue,
        focused,
        isCurrentDate:
          currentDate.getDate() === today.getDate() &&
          currentDate.getMonth() === today.getMonth() &&
          currentDate.getFullYear() === today.getFullYear()
      });
      weekDayIndex += 1;
      currentDate.setDate(currentDate.getDate() + 1);
    }

    return weeks as IWeekElement[];
  }

  private getFocusedValue(): Date {
    if (this.focusedValue) {
      return this.focusedValue;
    }
    return this.getSelectedValue();
  }

  private changeMonth(newMonthValue: number): Date {
    return new Date(
      this.getFocusedValue().getFullYear(),
      this.getFocusedValue().getMonth() + newMonthValue,
      1,
      0,
      0,
      0
    );
  }

  private setNewFocusedMonth(newMonthValue: number): void {
    this.focusedValue = this.changeMonth(newMonthValue);
  }

  private getSelectedValue(): Date {
    if (this.slottedInput.value) {
      const value = fromIsoDate(this.slottedInput.value);
      return value;
    }

    const now = new Date();
    now.setHours(0, 0, 0, 0);
    return now;
  }

  private renderHeader(): JSX.Element {
    return (
      <div class="gux-header">
        <button
          type="button"
          class="gux-left"
          aria-label={this.i18n('previousMonth', {
            localizedPreviousMonthAndYear: getDateAsMonthYear(
              this.changeMonth(-1),
              this.locale
            )
          })}
          onClick={() => this.setNewFocusedMonth(-1)}
        >
          <gux-icon decorative icon-name="chevron-small-left"></gux-icon>
        </button>
        <span class="gux-header-month-and-year">
          {getDateAsMonthYear(this.getFocusedValue(), this.locale)}
        </span>
        <button
          type="button"
          class="gux-right"
          aria-label={this.i18n('nextMonth', {
            localizedNextMonthAndYear: getDateAsMonthYear(
              this.changeMonth(1),
              this.locale
            )
          })}
          onClick={() => this.setNewFocusedMonth(1)}
        >
          <gux-icon decorative icon-name="chevron-small-right"></gux-icon>
        </button>
      </div>
    ) as JSX.Element;
  }

  private renderContent(): JSX.Element {
    return (
      <div>
        <div class="gux-week-days">
          {getWeekdays(this.locale, this.startDayOfWeek).map(
            day => (<div class="gux-week-day">{day}</div>) as JSX.Element
          )}
        </div>

        <div class="gux-content">
          {this.getMonthDays().map(
            week =>
              (
                <div class="gux-content-week">
                  {week.dates.map(
                    day =>
                      (
                        <div
                          data-date={day.date.getTime()}
                          onClick={() => this.onDateClick(day.date)}
                          role="button"
                          aria-current={day.selected ? 'true' : 'false'}
                          tabindex={day.selected || day.focused ? '0' : '-1'}
                          onKeyDown={e => void this.onKeyDown(e)}
                          aria-disabled={day.disabled ? 'true' : 'false'}
                          class={{
                            'gux-content-date': true,
                            'gux-disabled': day.disabled,
                            'gux-current-month': day.inCurrentMonth,
                            'gux-selected': day.selected,
                            'gux-current-date': day.isCurrentDate
                          }}
                        >
                          <span class="gux-non-sr" aria-hidden="true">
                            {day.date.getDate()}
                          </span>
                          <span class="gux-sr-only">
                            <gux-date-beta
                              datetime={day.date.toISOString()}
                              format="long"
                            ></gux-date-beta>
                          </span>
                        </div>
                      ) as JSX.Element
                  )}
                </div>
              ) as JSX.Element
          )}
        </div>
      </div>
    ) as JSX.Element;
  }

  render(): JSX.Element {
    return (
      <div class="gux-calendar-beta">
        <slot />
        {this.renderHeader()}
        {this.renderContent()}
      </div>
    ) as JSX.Element;
  }
}
