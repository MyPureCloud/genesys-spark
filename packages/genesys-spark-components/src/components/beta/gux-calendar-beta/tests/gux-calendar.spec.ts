import { newSpecPage } from '@test/specTestUtils';
import { GuxCalendar } from '../gux-calendar';

const components = [GuxCalendar];
const language = 'en';
const html = `
<gux-calendar-beta>
  <input type="date" value="2023-05-19" min="2023-04-28" max="2023-06-18" />
</gux-calendar-beta>
`;

describe('gux-calendar', () => {
  it('should build', async () => {
    const page = await newSpecPage({ components, html, language });

    expect(page.rootInstance).toBeInstanceOf(GuxCalendar);
  });

  it('renders', async () => {
    const page = await newSpecPage({ components, html, language });

    expect(page.root).toMatchSnapshot();
  });
});
