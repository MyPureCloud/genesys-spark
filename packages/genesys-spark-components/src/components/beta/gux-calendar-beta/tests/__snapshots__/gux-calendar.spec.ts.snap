// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-calendar renders 1`] = `
<gux-calendar-beta>
  <template shadowrootmode="open">
    <div class="gux-calendar-beta">
      <slot></slot>
      <div class="gux-header">
        <button aria-label="Previous month: April 2023" class="gux-left" type="button">
          <gux-icon decorative="" icon-name="custom/chevron-left-small-regular" size="small"></gux-icon>
        </button>
        <span class="gux-header-month-and-year">
          May 2023
        </span>
        <button aria-label="Next month: June 2023" class="gux-right" type="button">
          <gux-icon decorative="" icon-name="custom/chevron-right-small-regular" size="small"></gux-icon>
        </button>
      </div>
      <div>
        <div class="gux-content">
          <div class="gux-week-days">
            <div class="gux-week-day">
              S
            </div>
            <div class="gux-week-day">
              M
            </div>
            <div class="gux-week-day">
              T
            </div>
            <div class="gux-week-day">
              W
            </div>
            <div class="gux-week-day">
              T
            </div>
            <div class="gux-week-day">
              F
            </div>
            <div class="gux-week-day">
              S
            </div>
          </div>
          <div>
            <div class="gux-content-week">
              <gux-day-beta aria-current="false" class="day-2023-04-30 gux-muted" day="2023-04-30" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-05-01" day="2023-05-01" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-05-02" day="2023-05-02" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-05-03" day="2023-05-03" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-05-04" day="2023-05-04" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-05-05" day="2023-05-05" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-05-06" day="2023-05-06" tabindex="-1"></gux-day-beta>
            </div>
            <div class="gux-content-week">
              <gux-day-beta aria-current="false" class="day-2023-05-07" day="2023-05-07" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-05-08" day="2023-05-08" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-05-09" day="2023-05-09" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-05-10" day="2023-05-10" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-05-11" day="2023-05-11" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-05-12" day="2023-05-12" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-05-13" day="2023-05-13" tabindex="-1"></gux-day-beta>
            </div>
            <div class="gux-content-week">
              <gux-day-beta aria-current="false" class="day-2023-05-14" day="2023-05-14" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-05-15" day="2023-05-15" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-05-16" day="2023-05-16" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-05-17" day="2023-05-17" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-05-18" day="2023-05-18" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="true" class="day-2023-05-19" day="2023-05-19" tabindex="0"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-05-20" day="2023-05-20" tabindex="-1"></gux-day-beta>
            </div>
            <div class="gux-content-week">
              <gux-day-beta aria-current="false" class="day-2023-05-21" day="2023-05-21" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-05-22" day="2023-05-22" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-05-23" day="2023-05-23" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-05-24" day="2023-05-24" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-05-25" day="2023-05-25" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-05-26" day="2023-05-26" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-05-27" day="2023-05-27" tabindex="-1"></gux-day-beta>
            </div>
            <div class="gux-content-week">
              <gux-day-beta aria-current="false" class="day-2023-05-28" day="2023-05-28" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-05-29" day="2023-05-29" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-05-30" day="2023-05-30" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-05-31" day="2023-05-31" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-06-01 gux-muted" day="2023-06-01" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-06-02 gux-muted" day="2023-06-02" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-06-03 gux-muted" day="2023-06-03" tabindex="-1"></gux-day-beta>
            </div>
            <div class="gux-content-week">
              <gux-day-beta aria-current="false" class="day-2023-06-04 gux-muted" day="2023-06-04" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-06-05 gux-muted" day="2023-06-05" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-06-06 gux-muted" day="2023-06-06" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-06-07 gux-muted" day="2023-06-07" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-06-08 gux-muted" day="2023-06-08" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-06-09 gux-muted" day="2023-06-09" tabindex="-1"></gux-day-beta>
              <gux-day-beta aria-current="false" class="day-2023-06-10 gux-muted" day="2023-06-10" tabindex="-1"></gux-day-beta>
            </div>
          </div>
        </div>
      </div>
    </div>
  </template>
  <input max="2023-06-18" min="2023-04-28" type="date" value="2023-05-19">
</gux-calendar-beta>
`;
